### DISCUSSION TITLE
system-log-ttl not working.

### DISCUSSION THREAD

**Discussion by pk-lee-developer** [akvorado/akvorado#979] [Q&A]
In  'akvorado.yaml" file, 
      under "clickhouse:", 
           set "system-log-ttl: 0" does not work. 

Also tried set "system-logs-ttl: 0" does not work. 

* 1.7.0 - 2023-01-26
  orchestrator→clickhouse→system-logs-ttl to 0.

Also found this on  "akvorado-clickhouse-1" container set to 30 days.

root@320c00d2426f:/etc/clickhouse-server/config.d# cat akvorado.xml 
<clickhouse>
 <asynchronous_metric_log>
  <ttl>event_date + INTERVAL 2592000 SECOND DELETE</ttl>
 </asynchronous_metric_log>
 <metric_log>
  <ttl>event_date + INTERVAL 2592000 SECOND DELETE</ttl>
 </metric_log>
 <part_log>
  <ttl>event_date + INTERVAL 2592000 SECOND DELETE</ttl>
 </part_log>
 <query_log>
  <ttl>event_date + INTERVAL 2592000 SECOND DELETE</ttl>
 </query_log>
 <query_thread_log>
  <ttl>event_date + INTERVAL 2592000 SECOND DELETE</ttl>
 </query_thread_log>
 <trace_log>
  <ttl>event_date + INTERVAL 2592000 SECOND DELETE</ttl>
 </trace_log>
</clickhouse>

Any idea?

**Comment by vincentbernat** [comment]
`system-log-ttl` is indeed the correct value. Note that you need to restart the orchestrator. Also, recent versions of ClickHouse won't run the init script a second time. This will be fixed with https://github.com/akvorado/akvorado/commit/b8eeabc73ec5aa688cd7cf5dc08eceb223c6ffab.

**Reply by vincentbernat** [reply]
I have also released 1.9.2 for this specific issue.

**Comment by pk-lee-developer** [comment]
Upgraded to v1.9.2.

Setting "system-logs-ttl to 0" empty "akvorado.xml " now.

root@ddfa931cf39e:~# cat /etc/clickhouse-server/config.d/akvorado.xml 
>> <clickhouse>
>> </clickhouse>

TTL appears to be disabled for system log tables now.

docker exec akvorado-clickhouse-1 clickhouse-client \
--query "SELECT table,
             MIN(modification_time) AS oldest_modification_time,
             MIN(delete_ttl_info_max) AS oldest_delete_ttl_time
          FROM system.parts
          WHERE table LIKE '%log'
          GROUP by table
          ORDER BY table;" \
--format="Pretty"
┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ table                   ┃ oldest_modification_time ┃ oldest_delete_ttl_time ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━┩
│ asynchronous_metric_log │      2023-11-28 15:12:02 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ metric_log              │      2023-11-28 15:12:01 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ part_log                │      2023-11-28 15:12:01 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ query_log               │      2023-11-28 15:12:01 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ trace_log               │      2023-11-28 15:12:01 │    1970-01-01 00:00:00 │
└─────────────────────────┴──────────────────────────┴────────────────────────┘

System log tables still logging new data (rows).

docker exec akvorado-clickhouse-1 clickhouse-client \
--query "SELECT database, table,
             formatReadableSize(sum(data_compressed_bytes) AS size) AS compressed,
             formatReadableSize(sum(data_uncompressed_bytes) AS usize) AS uncompressed,
             round(usize / size, 2) AS compr_rate,
             sum(rows) AS rows
          FROM system.parts
          WHERE (active = 1) AND (database LIKE '%') AND (table LIKE '%')
          GROUP BY
            database,
            table
          ORDER BY (database,table);" \
--format="Pretty"
┏━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━┓
┃ database ┃ table                                          ┃ compressed ┃ uncompressed ┃ compr_rate ┃     rows ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━┩
│ default  │ .inner_id.cf22006e-8b3b-4322-bfc2-dba50b0811f7 │ 8.70 KiB   │ 13.75 KiB    │       1.58 │      166 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ default  │ flows                                          │ 976.48 MiB │ 9.55 GiB     │      10.02 │ 63664215 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ default  │ flows_1h0m0s                                   │ 84.63 MiB  │ 1004.16 MiB  │      11.87 │ 10629223 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ default  │ flows_1m0s                                     │ 289.50 MiB │ 3.18 GiB     │      11.23 │ 34414455 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ default  │ flows_5m0s                                     │ 310.48 MiB │ 3.51 GiB     │      11.58 │ 38051561 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ system   │ asynchronous_metric_log                        │ 19.49 KiB  │ 463.39 KiB   │      23.78 │    31248 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ system   │ metric_log                                     │ 120.90 KiB │ 528.58 KiB   │       4.37 │      113 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ system   │ part_log                                       │ 42.23 KiB  │ 248.36 KiB   │       5.88 │      540 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ system   │ query_log                                      │ 16.81 KiB  │ 83.64 KiB    │       4.98 │       64 │
├──────────┼────────────────────────────────────────────────┼────────────┼──────────────┼────────────┼──────────┤
│ system   │ trace_log                                      │ 30.90 KiB  │ 476.07 KiB   │      15.41 │     1380 │
└──────────┴────────────────────────────────────────────────┴────────────┴──────────────┴────────────┴──────────┘

 docker exec akvorado-clickhouse-1 clickhdocker exec akvorado-clickhouse-1 clickhouse-client \
--query "SELECT table,
            MAX(modification_time) AS newest_modification_time,
            MAX(delete_ttl_info_max) AS newest_delete_ttl_time
         FROM system.parts
         WHERE table LIKE '%log'
         GROUP by table
         ORDER BY table;" \
--format="Pretty"

┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ table                   ┃ newest_modification_time ┃ newest_delete_ttl_time ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━┩
│ asynchronous_metric_log │      2023-11-28 15:15:32 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ metric_log              │      2023-11-28 15:15:32 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ part_log                │      2023-11-28 15:15:31 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ query_log               │      2023-11-28 15:15:31 │    1970-01-01 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ trace_log               │      2023-11-28 15:15:31 │    1970-01-01 00:00:00 │
└─────────────────────────┴──────────────────────────┴────────────────────────┘

**Comment by pk-lee-developer** [comment]
Just a FYI...

Setting a custom TTL now works.

> system-log-ttl: 24h # 1 day

root@5caaefcc87cc:/etc/clickhouse-server/config.d# cat akvorado.xml 
<clickhouse>
 <asynchronous_metric_log>
  <ttl>event_date + INTERVAL 86400 SECOND DELETE</ttl>
 </asynchronous_metric_log>
 <metric_log>
  <ttl>event_date + INTERVAL 86400 SECOND DELETE</ttl>
 </metric_log>
 <part_log>
  <ttl>event_date + INTERVAL 86400 SECOND DELETE</ttl>
 </part_log>
 <query_log>
  <ttl>event_date + INTERVAL 86400 SECOND DELETE</ttl>
 </query_log>
 <query_thread_log>
  <ttl>event_date + INTERVAL 86400 SECOND DELETE</ttl>
 </query_thread_log>
 <trace_log>
  <ttl>event_date + INTERVAL 86400 SECOND DELETE</ttl>
 </trace_log>
</clickhouse>

docker exec akvorado-clickhouse-1 clickhouse-client \
--query "SELECT table,
             MIN(modification_time) AS oldest_modification_time,
             MIN(delete_ttl_info_max) AS oldest_delete_ttl_time
          FROM system.parts
          WHERE table LIKE '%log'
          GROUP by table
          ORDER BY table;" \
--format="Pretty"
┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ table                   ┃ oldest_modification_time ┃ oldest_delete_ttl_time ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━┩
│ asynchronous_metric_log │      2023-11-28 15:27:26 │    2023-11-29 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ metric_log              │      2023-11-28 15:27:26 │    2023-11-29 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ part_log                │      2023-11-28 15:27:26 │    2023-11-29 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ query_log               │      2023-11-28 15:27:26 │    2023-11-29 00:00:00 │
├─────────────────────────┼──────────────────────────┼────────────────────────┤
│ trace_log               │      2023-11-28 15:27:26 │    2023-11-29 00:00:00 │
└─────────────────────────┴──────────────────────────┴────────────────────────┘

**Comment by pk-lee-developer** [comment]
Just a recap.

Setting "system-logs-ttl to 0" only remove TTL for system log tables.  It does not disable logging.  It will keep system logs forever.

**Reply by vincentbernat** [reply]
Oh, yes, this was intended. This disables "setting the TTL".

**Comment by pk-lee-developer** [comment]
This is not my goal. I do not want the system logs to grow forever. 

I want all system logs disabled.

No sure if you want to implement this feature in future version,

This method below works for me.

1. Run inside ClickHouse.

cat \<\<EOT\>\> /etc/clickhouse-server/config.d/z_log_disable.xml
 \<?xml version="1.0"?\>
 \<clickhouse\>
     \<asynchronous_metric_log remove="1"\/>
     \<metric_log remove="1"\/>
     \<query_log remove="1" \/>
     \<part_log remove="1"\/>
     \<trace_log remove="1"\/>
 \</clickhouse\>
 EOT

2. Restart ClickHouse conrtainer.

3. Drop existing system logs.

docker exec akvorado-clickhouse-1 clickhouse-client \
--multiquery --query "DROP TABLE system.asynchronous_metric_log no delay;
                      DROP TABLE system.metric_log no delay;
                      DROP TABLE system.part_log no delay;
                      DROP TABLE system.query_log no delay;
                      DROP TABLE system.trace_log no delay;"

Changes are persistent and do not need to re-apply after each restarrt.

**Reply by vincentbernat** [reply]
No plan to implement that: the goal is not to manage every aspects of ClickHouse.

**Comment by pk-lee-developer** [comment]
Thats's fine. I posted the workaround in case anyone wants to disable them.  Thx
